codegen/post_gen_hooks.go

```diff
@@ -330,3 +341,162 @@ func writeAndFormat(path string, data []byte) error {
 	}
 	return FormatGoFile(path)
 }
+
+// WorkflowMockGenHook returns a PostGenHook to generate endpoint workflow mocks
+func WorkflowMockGenHook(h *PackageHelper, t *Template) PostGenHook {
+	return func(instances map[string][]*ModuleInstance) error {
+		errChanSize := 0
+		shouldGenMap := map[*ModuleInstance][]*EndpointSpec{}
+		for _, instance := range instances["endpoint"] {
+			endpointSpecs := instance.genSpec.([]*EndpointSpec)
+			for _, endpointSpec := range endpointSpecs {
+				if endpointSpec.WorkflowType == "custom" {
+					shouldGenMap[instance] = endpointSpecs
+					errChanSize += len(endpointSpecs)
+					break
+				}
+			}
+		}
+
+		genEndpointCount := len(shouldGenMap)
+		if genEndpointCount == 0 {
+			return nil
+		}
+		errChanSize += genEndpointCount
+
+		fmt.Printf("Generating %d endpoint mocks:\n", genEndpointCount)
+		ec := make(chan error, errChanSize)
+		var idx int32 = 1
+		var files sync.Map
+		var wg sync.WaitGroup
+		for instance, endpointSpecs := range shouldGenMap {
+			wg.Add(1)
+			go func(instance *ModuleInstance, endpointSpecs []*EndpointSpec) {
+				defer wg.Done()
+
+				buildDir := h.CodeGenTargetPath()
+				genDir := filepath.Join(buildDir, instance.Directory, "mock-workflow")
+
+				cwf, err := findClientsWithFixture(instance)
+				if err != nil {
+					ec <- errors.Wrapf(
+						err,
+						"Error generating mock endpoint %s",
+						instance.InstanceName,
+					)
+					return
+				}
+
+				var subWg sync.WaitGroup
```

Comment:
I wish go has something like `async.auto`

Reply:
Iol, I am glad that go does not have async.auto :p